tests/puppeteer/pages/BO/addBrand.js

```diff
@@ -0,0 +1,108 @@
+    this.enabledSwitchlabel = 'label[for=\'manufacturer_is_enabled_%ID\']';
+    // Selectors for Meta keywords
+    this.taggableFieldDiv = 'div.input-group div.js-locale-%LANG';
+    this.deleteKeywordLink = `${this.taggableFieldDiv} a.close`;
+    this.saveButton = '.card-footer button';
+  }
+
+  /*
+  Methods
+   */
+
+  /**
+   * Create or edit Brand
+   * @param brandData
+   * @return {Promise<void>}
+   */
+  async createEditBrand(brandData) {
+    // Fill Name
+    await this.setValue(this.nameInput, brandData.name);
+    // Fill information in english
+    await Promise.all([
+      this.page.click(this.shortDescriptionLangLink.replace('%LANG', 'en')),
+      this.page.waitForSelector(`${this.shortDescriptionLangLink.replace('%LANG', 'en')}.active`, {visible: true}),
+    ]);
+    await this.setValueOnTinymceInput(this.shortDescriptionIFrame.replace('%ID', '1'), brandData.shortDescription);
+    await this.setValueOnTinymceInput(this.descriptionIFrame.replace('%ID', '1'), brandData.description);
+    await this.setValueOnTinymceInput(this.metaTitleInput.replace('%ID', '1'), brandData.metaTitle);
+    await this.setValueOnTinymceInput(this.metaDescriptionInput.replace('%ID', '1'), brandData.metaDescription);
+    await this.deleteKeywords('en');
+    await this.addKeywords(brandData.metaKeywords, '1');
+
+    // Fill Information in french
+    await Promise.all([
+      this.page.click(this.shortDescriptionLangLink.replace('%LANG', 'fr')),
+      this.page.waitForSelector(`${this.shortDescriptionLangLink.replace('%LANG', 'fr')}.active`, {visible: true}),
+    ]);
+    await this.setValueOnTinymceInput(this.shortDescriptionIFrame.replace('%ID', '2'), brandData.shortDescriptionFr);
+    await this.setValueOnTinymceInput(this.descriptionIFrame.replace('%ID', '2'), brandData.descriptionFr);
+    await this.setValueOnTinymceInput(this.metaTitleInput.replace('%ID', '2'), brandData.metaTitleFr);
+    await this.setValueOnTinymceInput(this.metaDescriptionInput.replace('%ID', '2'), brandData.metaDescriptionFr);
+    await this.deleteKeywords('fr');
+    await this.addKeywords(brandData.metaKeywordsFr, '2');
+
+    // Add logo
+    await this.generateAndUploadImage(this.logoFileInput, brandData.logo);
+
+    // Set Enabled value
+    if (brandData.enabled) {
+      await this.page.click(this.enabledSwitchlabel.replace('%ID', '1'));
+    } else {
+      await this.page.click(this.enabledSwitchlabel.replace('%ID', '0'));
+    }
+    // Save Created brand
+    await this.clickAndWaitForNavigation(this.saveButton);
+    await this.page.waitForSelector(this.alertSuccessBlockParagraph, {visible: true});
+    return this.getTextContent(this.alertSuccessBlockParagraph);
+  }
+
+  /**
+   * Delete all keywords
+   * @param lang, to specify which input to empty
+   * @return {Promise<void>}
+   */
+  async deleteKeywords(lang = 'en') {
+    const closeButtons = await this.page.$$(this.deleteKeywordLink.replace('%LANG', lang));
```

Comment:
Might be a good idea to `trim()` and `.toLowerCase()` the `lang` parameter ?
Just to be sure

Reply:
we don't have to tolerate incorrect strings for lang, 
Lang is used by a selector and test should fail if language is wrong.